text/0232-maturity-of-otel.md

```diff
@@ -0,0 +1,151 @@
+
+The Collector's [stability levels](https://github.com/open-telemetry/opentelemetry-collector#stability-levels) inspired the maturity levels.
+
+## Motivation
+
+Quite often, the community is faced with the question of the quality and maturity expectations of its diverse set of components. This OTEP aims to bring clarity by establishing a framework to communicate the project's maturity. As the OpenTelemetry project comprises a multitude of SIGs, and each SIG has several components of varying quality, having this framework will help set the right expectations for OpenTelemetry users.
+
+Only once we reach the "stable" level at the project level will we be ready to consider graduating OpenTelemetry at the CNCF, with all of the Tier-1 SIGs being in the context of the graduation, including their Tier-1 Components. This OTEP provides a framework for describing project maturity as a whole.
+
+## Glossary
+
+* *SIG:* an official SIG within the OpenTelemetry project, like the Collector or Java. Some WGs are temporarily accepted, given they should be SIGs instead of WGs, like the Semantic Conventions WG.
+* *Tier-1 SIG:* a SIG within the OpenTelemetry project deemed critical by the GC/TC.
+* *Candidate Tier-1 SIG:* a SIG that is deemed as critical for the future of the project, but might not be of the same maturity level yet. It's expected to reach that level eventually, and other Tier-1 SIGs should be prepared to accommodate the new component as needed.
+* *Component:* module maintained by a SIG like a Collector exporter or an SDK sampler.
+* *Tier-1 Component:* a component selected by project maintainers as critical for the SIG. For instance, the OTLP receiver will likely be selected as a Tier-1 Component of the Collector. See Tier-1 Components for more details.
+* *Candidate Tier-1 Component:* similar to "Candidate Tier-1 SIG", but for a component within a SIG.
+* *Dependency:* an OpenTelemetry SIG or component that is required by another SIG or component. For instance, the OTLP Logs Data Model from the Spec is required for the Collector's Logging support.  
+
+## Explanation
+
+### Maturity levels
+
+SIGs and components of a project MUST have a declared maturity level established by SIG maintainers (SIGs). The maturity level for a SIG or component is, at most, the lowest level of its dependencies or priority components. For instance, if the Collector SIG maintainers declare the "otlpreceiver" component as a tier-1 component of the Collector core distribution, and the "otlpreceiver" declares a dependency on the OpenTelemetry Collector API "config" package which is marked with a maturity level of "beta", the "otlpreceiver" can be at most "beta". The OpenTelemetry Collector "core" distribution is also affected so that it can be at most "beta".
```

Comment:
I don't see a conflict, as the `otlpreceiver` itself can list which parts of it are considered tier-1. Should the OTLP receiver support a new signal tomorrow that is not a tier-1 signal yet, we'd certainly want to get that feature out while not marking it as "tier-1". Similarly, the Collector SIG can declare that only the config is part of the deliverable for the components, and not their internal APIs.

Reply:
My problem is not with the component having different stability levels for different signals, the problem I see is that the OTLP traces receiver is marked as stable while a dependency (the `config` package) is, (pressumably, since it's at `0.x`) not stable. 

So, one of the following must be true:
1. The current stability level for the OTLP traces receiver is wrong
2. The OTLP traces receiver does not "really" take a dependency on the `config` package
3. The rule *Only once all dependencies are marked as "stable" MAY a component be marked as "stable"* should not be followed

Which one is it?

Reply:
IMO, it's the first option. It shouldn't have been marked stable if an underlying component isn't stable yet. What if the underlying config changes?

Reply:
See open-telemetry/opentelemetry-collector/issues/7964, let's continue the discussion there.

Reply:
I'm marking this as resolved, as the discussion has been moved there.